### PR TITLE
feat: add nightly minimum dependencies test

### PR DESCRIPTION
**Pull request by jalil-salame**
Add a nightly toolchain to test builds using the `-Z direct-minimum-dependencies` flag. This ensures our minimum version requirements are compatible with our source code.

This required bumping some versions:

- libc: `0.2.138` -> `0.2.158` as `vsock-rs` depends on `0.2.158`.
- tokio: `1` -> ~~`1.27` as we depend on methods/trait impls which are only available starting that version.~~ `1.34` as tonic 0.12 requires tokio 1.34

This ensures our minimum `tonic` requirement still compiles.

This should be released as `0.6.1` as requested by @cryo28 on #53

### CODE REVIEW

**Review by qwandor — APPROVED** [review]

**Review by Tim-Zhang — APPROVED** [review]
LGTM, thanks @jalil-salame

**Comment by Tim-Zhang** [comment]
@jalil-salame I will cut a release after this pr merged

**Comment by jalil-salame** [comment]
I'll do a small rework cause the tests no longer work here (tonic-conn feature was removed)

**Comment by jalil-salame** [comment]
I adjusted the branch rules, so that it now requires `BVT (stable)` and `BVT (nightly)` to pass (instead of just `BVT`).

**Review by qwandor — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -28,7 +29,7 @@ tonic012 = { package = "tonic", version = "0.12", optional = true }
 [dev-dependencies]
 sha2 = "0.10.6"
 rand = "0.8.5"
-tokio = { version = "1", features = ["macros", "rt", "io-util"] }
+tokio = { version = "1.34", features = ["macros", "rt", "io-util"] }
```
Comment (Tim-Zhang) [inline Cargo.toml:32]: Why don't use 1.27 wrote in the description? In my opinion, use a lower version will have less impact on user code.

**Comment by Tim-Zhang** [comment]
@jalil-salame I left a comment, PTAL, thanks.

File: Cargo.toml (hunk 1 above)
Comment (jalil-salame) [reply Cargo.toml:32]: Tonic 0.12 requires tokio 1.34, so 1.27 is incompatible with the tonic012 feature
